tests/test_merkle_drop.py

```diff
@@ -40,3 +63,64 @@ def test_incorrect_proof_entitlement(
         ).call()
         is False
     )
+
+
+def test_withdraw(
+    merkle_drop_contract, tree_data, proofs_for_tree_data, dropped_token_contract
+):
+    for i in range(len(proofs_for_tree_data)):
+        merkle_drop_balance = dropped_token_contract.functions.balanceOf(
+            merkle_drop_contract.address
+        ).call()
+
+        address = tree_data[i].address
+        value = tree_data[i].value
+        proof = proofs_for_tree_data[i]
+        merkle_drop_contract.functions.withdrawFor(address, value, proof).transact()
+
+        assert dropped_token_contract.functions.balanceOf(address).call() == value
+        assert (
+            dropped_token_contract.functions.balanceOf(
+                merkle_drop_contract.address
+            ).call()
+            == merkle_drop_balance - value
+        )
+
+
+def test_withdraw_already_withdrawn(
+    merkle_drop_contract_already_withdrawn,
+    eligible_address_0,
+    eligible_value_0,
+    proof_0,
+):
+    with pytest.raises(eth_tester.exceptions.TransactionFailed):
+        merkle_drop_contract_already_withdrawn.functions.withdrawFor(
+            eligible_address_0, eligible_value_0, proof_0
+        ).transact()
+
+
+def test_withdraw_wrong_proof(
+    merkle_drop_contract_already_withdrawn, other_data, proof_0
+):
+    with pytest.raises(eth_tester.exceptions.TransactionFailed):
+        merkle_drop_contract_already_withdrawn.functions.withdrawFor(
+            other_data[0].address, other_data[0].value, proof_0
+        ).transact()
+
+
+def test_withdraw_event(
+    merkle_drop_contract, web3, eligible_address_0, eligible_value_0, proof_0
+):
+
+    latest_block_number = web3.eth.blockNumber
+
+    merkle_drop_contract.functions.withdrawFor(
+        eligible_address_0, eligible_value_0, proof_0
+    ).transact()
+
+    event = merkle_drop_contract.events.Withdraw.createFilter(
+        fromBlock=latest_block_number
```

Comment:
The `fromBlock` should become obsolete in case you decide for a `session` scoped contract fixture (above conversation).

Reply:
It depends on the side effects of other session scoped fixtures.